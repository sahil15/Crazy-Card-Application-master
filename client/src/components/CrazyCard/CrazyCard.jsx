import React from "react";
import DisplayCard from "../DisplayCard/DisplayCard.jsx";
import PropTypes from "prop-types";

class CrazyCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hideAddBtn: false,
      hideRemoveBtn: true,
    };
  }
  render() {

    const selectCard = () => {
      this.setState({
        hideAddBtn: true,
        hideRemoveBtn: false,
      });
      this.props.onSelect(this.props.card);
    };
    const removeCard = () => {
      this.props.onRemove(this.props.card);
      this.setState({
        hideAddBtn: false,
        hideRemoveBtn: true,
      });
    };

    return (
      <DisplayCard
        card={this.props.card}
        hideAddBtn={this.state.hideAddBtn}
        hideRemoveBtn={this.state.hideRemoveBtn}
        selectCard= {selectCard}
        removeCard={removeCard}
      ></DisplayCard>
    
    );
  }
}

CrazyCard.propTypes = {
  card:PropTypes.object,
  onSelect:PropTypes.func,
  onRemove:PropTypes.func,

} 
export default CrazyCard;
