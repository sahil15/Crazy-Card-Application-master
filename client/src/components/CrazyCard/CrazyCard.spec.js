import React from 'react';
import { shallow } from 'enzyme';
import CrazyCard from './CrazyCard.jsx';

describe("Tests for Crazy Card", ()=> {
    const mockCard = {
        "name": "Student Life Card",
        "image": "https://ddialhyn49dxu.cloudfront.net/cards/hsbc2018-large.png",
        "apr": "18.9%",
        "balanceOfferDuration": "0 months",
        "purchaseOfferDuration": "6 months",
        "credit": "Â£1200",
        "min_income": 0,
        "empStatus": "Student"
        };
    it('Renders correctly', () => {
        const wrapper = shallow(<CrazyCard card={mockCard} />);
        expect(wrapper).toMatchSnapshot();
      });
})

   
   

