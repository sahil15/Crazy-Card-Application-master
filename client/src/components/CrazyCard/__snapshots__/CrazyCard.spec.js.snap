// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Crazy Card Renders correctly 1`] = `
<DisplayCard
  card={
    Object {
      "apr": "18.9%",
      "balanceOfferDuration": "0 months",
      "credit": "Â£1200",
      "empStatus": "Student",
      "image": "https://ddialhyn49dxu.cloudfront.net/cards/hsbc2018-large.png",
      "min_income": 0,
      "name": "Student Life Card",
      "purchaseOfferDuration": "6 months",
    }
  }
  hideAddBtn={false}
  hideRemoveBtn={true}
  removeCard={[Function]}
  selectCard={[Function]}
/>
`;
